python/ray/serve/config.py

```diff
@@ -17,86 +19,59 @@ def _callable_is_blocking(func_or_class):
         return not inspect.iscoroutinefunction(func_or_class.__call__)
 
 
-class BackendConfig:
-    def __init__(self, config_dict, accepts_batches=False, is_blocking=True):
-        assert isinstance(config_dict, dict)
-        # Make a copy so that we don't modify the input dict.
-        config_dict = config_dict.copy()
-
-        self.accepts_batches = accepts_batches
-        self.is_blocking = is_blocking
-        self.num_replicas = config_dict.pop("num_replicas", 1)
-        self.max_batch_size = config_dict.pop("max_batch_size", None)
-        self.batch_wait_timeout = config_dict.pop("batch_wait_timeout", 0)
-        self.max_concurrent_queries = config_dict.pop("max_concurrent_queries",
-                                                      None)
-        self.autoscaling_config = config_dict.pop("autoscaling", None)
-
-        if self.max_concurrent_queries is None:
+class BackendConfig(BaseModel):
+    accepts_batches: bool = False
```

Comment:
One issue I see here is that I don't actually think we want this as a user-facing option in the config. This is just used to pipe through that the backend accepts batches to the backend (the API is to use the `serve.accepts_batch` decorator). We could also expose it in the `BackendConfig` as an alternative, I'm not too opinionated. @simon-mo what do you think?

Reply:
I think it's fine to expose those as read only member variables (so it's useful for diagnostic and debugging). We can prefix them with `_` to indicate they are internal use.